pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -0,0 +1,109 @@
+            "hass-enforce-type-hints-config-entry",
+            "Used when setup has some arguments typed but first isn't ConfigEntry",
+        ),
+        "W0015": (
+            "Return type should be of type bool",
+            "hass-enforce-type-hints-return-bool",
+            "Used when setup has some arguments typed but doesn't return bool",
+        ),
+        "W0016": (
+            "First argument should be of type HomeAssistant",
+            "hass-enforce-type-hints-return-none",
+            "Used when setup has some arguments typed but doesn't return None",
+        ),
+    }
+    options = ()
+
+    def visit_functiondef(self, node: astroid.FunctionDef) -> None:
+        """Called when a FunctionDef node is visited."""
+        if node.is_method() or node.name not in [
+            "setup",
+            "async_setup",
+            "async_setup_entry",
+            "async_remove_entry",
+            "async_unload_entry",
+            "async_migrate_entry",
+        ]:
+            return
+
+        # Check that all arguments are annotated.
+        # The first argument is "self".
+        args = node.args
+        annotations = (
+            args.posonlyargs_annotations
+            + args.annotations
+            + args.kwonlyargs_annotations
+        )
+        if args.vararg is not None:
+            annotations.append(args.varargannotation)
+        if args.kwarg is not None:
+            annotations.append(args.kwargannotation)
+        if not [annotation for annotation in annotations if annotation is not None]:
+            return
+
+        print(annotations)
+
+        # Check that the first argument is "HomeAssistant".
+        if (
+            not isinstance(annotations[0], astroid.Name)
+            or annotations[0].name != "HomeAssistant"
+        ):
+            self.add_message("hass-enforce-type-hints-hass", node=node)
+        # Check that the second argument is "ConfigType" or "ConfigEntry".
+        if node.name in ["setup", "async_setup"]:
+            if (
+                not isinstance(annotations[1], astroid.Name)
+                or annotations[1].name != "ConfigType"
+            ):
+                self.add_message("hass-enforce-type-hints-config-type", node=node)
+        else:
+            if (
+                not isinstance(annotations[1], astroid.Name)
+                or annotations[1].name != "ConfigEntry"
+            ):
+                self.add_message("hass-enforce-type-hints-config-entry", node=node)
+        # Check that return type is specified and it is "None".
+        if node.name == "async_remove_entry":
+            if (
+                not isinstance(node.returns, astroid.Const)
+                or node.returns.value is not None
+            ):
+                self.add_message("hass-enforce-type-hints-return-none", node=node)
+        else:
+            if (
+                not isinstance(node.returns, astroid.Name)
+                or node.returns.name != "bool"
```

Comment:
Are you sure `bool` is represented this way?

Reply:
That's what it came up with: `Name.bool(name='bool')`